tests/raftstore/test_single.rs

```diff
@@ -59,6 +62,33 @@ fn test_delete<T: Simulator>(cluster: &mut Cluster<T>) {
     }
 }
 
+fn test_wrong_store_id<T: Simulator>(cluster: &mut Cluster<T>) {
+    cluster.run();
+
+    let (k, v) = (b"k", b"v");
+    for _ in 0..20 {
+        let mut region = cluster.get_region(k);
+        let region_id = region.get_id();
+        let cmd = new_put_cmd(k, v);
+        let mut req = new_request(region_id, region.take_region_epoch(), vec![cmd], true);
+        let mut leader = cluster.leader_of_region(region_id).unwrap();
+        // setup wrong store id.
+        let store_id = leader.get_store_id();
+        leader.set_store_id(store_id + 1);
+        req.mut_header().set_peer(leader);
+        let result = cluster.call_command(store_id, req, Duration::from_secs(5));
+        if let Err(Error::Timeout(_)) = result {
```

Comment:
@BusyJay Tried to remove the error check in `cluster.rs`, and it fails these tests:
```
failures:
    raftstore::test_conf_change::test_node_split_brain
    raftstore::test_conf_change::test_server_split_brain
    raftstore::test_transport::test_node_partition_write
    raftstore::test_transport::test_server_partition_write
```